controllers/managedfusion_controller.go

```diff
@@ -508,6 +510,13 @@ func (r *ManagedFusionReconciler) reconcileAlertmanagerConfig() error {
 		if err != nil {
 			return fmt.Errorf("unable to read customernotification.html file: %v", err)
 		}
+		smtpURL, err := url.ParseRequestURI(r.smtpConfigData.Endpoint)
+		if err != nil {
+			return fmt.Errorf("unable to parse SMTP Endpoint url: %v", err)
+		}
+		if smtpURL.Hostname() == "" {
+			return fmt.Errorf("cannot parse host in smtp endpoint")
```

Comment:
You are not trying to retrieve the "host" (which is `hostname:port`), you are trying to retrieve the hostname
```suggestion
			return fmt.Errorf("cannot retrieve hostname from %s, might be a malformed or partial URL", r.smtpConfigData)
```